base/setup/lib/utils/devutils.c

```diff
@@ -66,11 +93,37 @@ pOpenDeviceEx(
  *
  * @param[in]   DevicePath
  * @param[out]  DeviceHandle
- * See the DevicePath and DeviceHandle parameters of pOpenDeviceEx().
+ * See the DevicePath and DeviceHandle parameters of pOpenDeviceEx_UStr().
  *
  * @return  An NTSTATUS code indicating success or failure.
  *
- * @see pOpenDeviceEx()
+ * @see pOpenDevice(), pOpenDeviceEx(), pOpenDeviceEx_UStr()
+ **/
+NTSTATUS
+pOpenDevice_UStr(
+    _In_ PCUNICODE_STRING DevicePath,
+    _Out_ PHANDLE DeviceHandle)
+{
+    return pOpenDeviceEx_UStr(DevicePath,
+                              DeviceHandle,
+                              FILE_READ_DATA | FILE_READ_ATTRIBUTES,
+                              FILE_SHARE_VALID_FLAGS // FILE_SHARE_READ,WRITE,DELETE
```

Comment:
I'm not a big fan of using a validation mask as flags even though this mask is unlikely to change in the future.

Reply:
indeed (and as you suspected it was to avoid writing the whole read|write|delete thing)

Reply:
`FILE_SHARE_VALID_FLAGS & 7` 🤣

Reply:
Lol that's worse 😂 well here I want all the valid flags to be present, so.... the define name is actually quite suitable!

Reply:
How do you know you want all flags? You want the 3 you know about. What if in the future MS invents FILE_SHARE_IGNOREOPLOCKS or FILE_SHARE_POSIXSEMANTICS? That is why using a validation mask is bad.

Reply:
I'm going to add a `FILE_SHARE_RWD` define in devutils.h, that's going to be the OR'ed of the corresponding read, write, delete flags.